File: completed/2044.java
import java.util.*;

class Solution {
    private int find(int[] nums, int index, int current, int maximum) {
        if (index == nums.length) {
            return 0;
        }

        int skip = find(nums, index + 1, current, maximum);
        current |= nums[index];
        int take = find(nums, index + 1, current, maximum);
        return skip + take + (current == maximum ? 1 : 0);
    }

    public int countMaxOrSubsets(int[] nums) {
        int maximum = 0;
        for (int num : nums) {
            maximum |= num;
        }
        return find(nums, 0, 0, maximum);
    }
}

File: completed/2044.cpp
class Solution {
private:
    int countSubsets(
        vector<int>& nums,
        int index,
        int current,
        int target
    ) {
        if (current == target) {
            return 1 << (nums.size() - index);
        }

        if (index == nums.size()) {
            return 0;
        }

        int take = countSubsets(nums, index + 1, current | nums[index], target);
        int skip = countSubsets(nums, index + 1, current, target);
        return take + skip;
    }

public:
    int countMaxOrSubsets(vector<int>& nums) {
        int target = 0;
        for (int num : nums) {
            target |= num;
        }
        return countSubsets(nums, 0, 0, target);
    }
};
